packages/react-hooks/src/useInflation.ts

```diff
@@ -7,35 +7,49 @@ import type { Inflation } from './types';
 
 import { useEffect, useState } from 'react';
 
-import { getInflationParams } from '@polkadot/apps-config';
-import { BN_MILLION, BN_ZERO } from '@polkadot/util';
+import { getInflationParams, UniformEraPayoutInflationParams } from '@polkadot/apps-config';
+import { BN_BILLION, BN_MILLION, BN_ZERO } from '@polkadot/util';
+import { BN_THOUSAND } from '@polkadot/util/bn/consts';
 
 import { createNamedHook } from './createNamedHook';
 import { useApi } from './useApi';
 import { useCall } from './useCall';
 
 const EMPTY: Inflation = { idealInterest: 0, idealStake: 0, inflation: 0, stakedFraction: 0, stakedReturn: 0 };
 
+function calcInflationUniformEraPayout (totalIssuance: BN, inflationParams: UniformEraPayoutInflationParams): number {
+  const totalIssuanceInTokens = totalIssuance.div(BN_BILLION).div(BN_THOUSAND);
+
+  return (totalIssuanceInTokens.isZero() ? 0.0 : inflationParams.yearlyInflationInTokens.toNumber() / totalIssuanceInTokens.toNumber());
+}
+
 function calcInflation (api: ApiPromise, totalStaked: BN, totalIssuance: BN, numAuctions: BN): Inflation {
-  const { auctionAdjust, auctionMax, falloff, maxInflation, minInflation, stakeTarget } = getInflationParams(api);
+  const inflationParams = getInflationParams(api);
+  const { auctionAdjust, auctionMax, falloff, maxInflation, minInflation, stakeTarget } = inflationParams;
   const stakedFraction = totalStaked.isZero() || totalIssuance.isZero()
     ? 0
     : totalStaked.mul(BN_MILLION).div(totalIssuance).toNumber() / BN_MILLION.toNumber();
   const idealStake = stakeTarget - (Math.min(auctionMax, numAuctions.toNumber()) * auctionAdjust);
   const idealInterest = maxInflation / idealStake;
-  const inflation = 100 * (minInflation + (
-    stakedFraction <= idealStake
-      ? (stakedFraction * (idealInterest - (minInflation / idealStake)))
-      : (((idealInterest * idealStake) - minInflation) * Math.pow(2, (idealStake - stakedFraction) / falloff))
-  ));
+  let inflationInPercentage = 0;
+
+  if ('yearlyInflationInTokens' in inflationParams) {
+    inflationInPercentage = 100 * calcInflationUniformEraPayout(totalIssuance, inflationParams as UniformEraPayoutInflationParams);
+  } else {
+    inflationInPercentage = 100 * (minInflation + (
```

Comment:
can we refactor this to a similar fn (e.g. `calculateInflationRewardCurve`)?

Reply:
Sure, will do